src/DCON/Ode.jl

```diff
@@ -0,0 +1,722 @@
+
+"""
+
+function ode_axis_init(ising::Int, ctrl::DconControl, equil::Equilibrium.PlasmaEquilibrium, intr::DconInternal, odet::OdeState)
+
+    # This might be wrong - double check once equil is more defined. 
+    qval(psi) = JPEC.SplinesMod.spline_eval(equil.sq, psi, 0)[4]
+    q1val(psi) = JPEC.SplinesMod.spline_eval(equil.sq, psi, 1)[2][4]
+
+    # Preliminary computations
+    odet.psifac = equil.sq.xs[1]  
+
+    # Use Newton iteration to find starting psi if qlow is above q0
+    # TODO: add this. For now, assume qlow = 0.0 and start at the first singular surface
+    if ctrl.qlow > 0.0
+        @warn "qlow > 0.0 is not implemented in ode_axis_init yet"
+    end
+
+    # Find inner singular surface
+    ising = 0
+    if false #(TODO: kin_flag)
+        # for ising = 1:kmsing
+        #     if kinsing[ising].psifac > psifac
+        #         break
+        #     end
+        # end
+    else
+        for ising in 1:intr.msing
+            if intr.sing[ising].psifac > odet.psifac
+                break
+            end
+        end
+    end
+    ising = max(0, ising - 1)
+
+    # Find next singular surface
+    if false # TODO: (kin_flag)
+        # while true
+        #     ising += 1
+        #     if ising > kmsing
+        #         break
+        #     end
+        #     if psilim < kinsing[ising].psifac
+        #         break
+        #     end
+        #     odet.q = kinsing[ising].q
+        #     if mlow <= nn * odet.q && mhigh >= nn * odet.q
+        #         break
+        #     end
+        # end
+        # if ising > kmsing || singfac_min == 0
+        #     psimax = psilim * (1 - eps)
+        #     next = "finish"
+        # elseif psilim < kinsing[ising].psifac
+        #     psimax = psilim * (1 - eps)
+        #     next = "finish"
+        # else
+        #     psimax = kinsing[ising].psifac - singfac_min / abs(nn * kinsing[ising].q1)
+        #     next = "cross"
+        # end
+    else
+        while true
+            ising += 1
+            if ising > intr.msing || intr.psilim < intr.sing[min(ising, msing)].psifac
+                break
+            end
+            odet.q = intr.sing[ising].q
+            if intr.mlow <= ctrl.nn * odet.q && intr.mhigh >= ctrl.nn * odet.q
+                break
+            end
+        end
+        if ising > intr.msing || intr.psilim < intr.sing[min(ising, intr.msing)].psifac || ctrl.singfac_min == 0
+            psimax = psilim * (1 - eps)
+            odet.next = "finish"
+        else
+            psimax = intr.sing[ising].psifac - ctrl.singfac_min / abs(ctrl.nn * intr.sing[ising].q1)
+            odet.next = "cross"
+        end
+    end
+
+    # Allocate and sort solutions by increasing value of |m-ms1|
+    m = intr.mlow - 1 .+ collect(1:intr.mpert)
+    if ctrl.sort_type == "absm"
+        key = abs.(m)
+    elseif ctrl.sort_type == "sing"
+        key = m
+        if intr.msing > 0
+            key = key .- intr.sing[1].m
+        end
+        key = -abs.(key)
```

Comment:
This is a tricky one! The minus is actually an unbroadcasted operation here (as is the assignment), so you're making a new array for `abs.(key)`, making another new array for the `-`, then assigning key to the new array. You want:
 `key .= .-abs.(key)`
or 
`@. key = -abs(key)`